### ISSUE TITLE
Support configurable hit test behavior on DragTarget

### ISSUE THREAD

**Issue by adabru** [flutter/flutter#73331]
Conceptually related issue: https://github.com/flutter/flutter/issues/73330

## Steps to Reproduce

1. Tried on version 1.26.
1. Run `flutter create issue`.
2. Run `cd issue`.
3. Update the file `lib/main.dart` as follows:

<details><summary>Code</summary>

    ```dart
    import 'dart:math';

    import 'package:flutter/gestures.dart';
    import 'package:flutter/material.dart';
    import 'package:flutter/rendering.dart';

    void main() {
      debugPaintSizeEnabled = true;
      runApp(DragTargetIssue());
    }

    class DragTargetIssue extends StatefulWidget {
      @override
      DragTargetIssueState createState() => DragTargetIssueState();
    }

    class DragTargetIssueState extends State<DragTargetIssue> {
      bool _isDraggedOver = false;
      bool _isTapped = false;
      @override
      Widget build(BuildContext context) {
        return MaterialApp(
          home: Scaffold(
            body: Column(
              children: [
                SizedBox(
                  width: 200,
                  height: 200,
                  child: DragTarget(
                    onMove: (details) {
                      _isDraggedOver = true;
                    },
                    onLeave: (details) => setState(() {
                      _isDraggedOver = false;
                    }),
                    builder: (context, candidateData, rejectedData) =>
                        GestureDetector(
                      onTap: () => setState(() {
                        _isTapped = !_isTapped;
                      }),
                      child: CustomPaint(
                        painter: _CustomPainter(
                          color: Color(0xFF000000)
                              .withRed(_isDraggedOver ? 255 : 0)
                              .withBlue(_isTapped ? 255 : 0),
                        ),
                        child: null,
                      ),
                    ),
                  ),
                ),
                Draggable(
                  feedback: Text('Dragging'),
                  child: Text('Drag me over the circle!'),
                ),
              ],
            ),
          ),
        );
      }
    }

    class _CustomPainter extends CustomPainter {
      final Color color;
      _CustomPainter({this.color});

      @override
      bool shouldRepaint(covariant CustomPainter oldDelegate) {
        return true;
      }

      @override
      void paint(Canvas canvas, Size size) {
        final Paint p = Paint();
        p.color = color;
        canvas.drawCircle(Offset(100, 100), 80, p);
      }

      @override
      bool hitTest(Offset position) {
        return pow(position.dx - 100, 2) + pow(position.dy - 100, 2) < pow(80, 2);
      }
    }
    ```
</details>

4. Run `flutter run -d linux` (or any other platform).
5. Drag the text `Drag me over the circle!` over the circle and the circle's bounding box outside the circle. In both cases the circle will turn red.
6. Click with the mouse on the circle and on the circle's bounding box outside the circle. When clicking the circle it will turn blue, when clicking outside the circle, nothing will happen.

**Expected result:** Both, dragging and click, should only work inside the circle, where the given `hitTest` function returns true.

**Actual result:** Clicking outside the circle doesn't register a tap as exptected. But dragging outside the circle is being registered which is not exptected.

**Comment:** Unlike issue https://github.com/flutter/flutter/issues/73330, I couldn't find the responsible code in this case. I stopped at the line https://github.com/flutter/flutter/blob/1.26.0-1.0.pre/packages/flutter/lib/src/widgets/drag_target.dart#L784, where `_DragAvatar` apparently performs its hitTest, but I didn't understand where the RenderObjects' hitTest functions are called.

## Logs

<details>
<summary>Logs</summary>

```
[  +90 ms] executing: [/opt/flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[  +42 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] 63062a64432cce03315d6b5196fda7912866eb37
[   +2 ms] executing: [/opt/flutter/] git tag --points-at 63062a64432cce03315d6b5196fda7912866eb37
[  +27 ms] Exit code 0 from: git tag --points-at 63062a64432cce03315d6b5196fda7912866eb37
[        ] 1.26.0-1.0.pre
[  +51 ms] executing: [/opt/flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[   +8 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/dev
[        ] executing: [/opt/flutter/] git ls-remote --get-url origin
[   +7 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[  +85 ms] executing: [/opt/flutter/] git rev-parse --abbrev-ref HEAD
[   +6 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] dev
[  +72 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +3 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +83 ms] Artifact Instance of 'MaterialFonts' is not required, skipping update.
[        ] Artifact Instance of 'GradleWrapper' is not required, skipping update.
[        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] Artifact Instance of 'FlutterSdk' is not required, skipping update.
[        ] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FontSubsetArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'PubDependencies' is not required, skipping update.
[  +86 ms] Creating project issue...
[  +55 ms]   issue/.idea/libraries/Dart_SDK.xml (created)
[  +22 ms]   issue/.idea/libraries/KotlinJavaRuntime.xml (created)
[        ]   issue/.idea/modules.xml (created)
[   +1 ms]   issue/.idea/runConfigurations/main_dart.xml (created)
[        ]   issue/.idea/workspace.xml (created)
[   +1 ms]   issue/.metadata (created)
[   +1 ms]   issue/README.md (created)
[   +1 ms]   issue/android/app/src/main/kotlin/com/example/issue/MainActivity.kt (created)
[   +1 ms]   issue/android/app/build.gradle (created)
[   +1 ms]   issue/android/issue_android.iml (created)
[   +1 ms]   issue/android/build.gradle (created)
[   +2 ms]   issue/android/.gitignore (created)
[   +1 ms]   issue/android/app/src/debug/AndroidManifest.xml (created)
[   +1 ms]   issue/android/app/src/main/res/drawable/launch_background.xml (created)
[        ]   issue/android/app/src/main/res/mipmap-hdpi/ic_launcher.png (created)
[        ]   issue/android/app/src/main/res/mipmap-mdpi/ic_launcher.png (created)
[        ]   issue/android/app/src/main/res/mipmap-xhdpi/ic_launcher.png (created)
[        ]   issue/android/app/src/main/res/mipmap-xxhdpi/ic_launcher.png (created)
[        ]   issue/android/app/src/main/res/mipmap-xxxhdpi/ic_launcher.png (created)
[        ]   issue/android/app/src/main/res/values/styles.xml (created)
[        ]   issue/android/app/src/main/res/values-night/styles.xml (created)
[        ]   issue/android/app/src/main/res/drawable-v21/launch_background.xml (created)
[        ]   issue/android/app/src/main/AndroidManifest.xml (created)
[   +1 ms]   issue/android/app/src/profile/AndroidManifest.xml (created)
[        ]   issue/android/gradle/wrapper/gradle-wrapper.properties (created)
[        ]   issue/android/gradle.properties (created)
[        ]   issue/android/settings.gradle (created)
[        ]   issue/ios/Runner/AppDelegate.swift (created)
[        ]   issue/ios/Runner/Runner-Bridging-Header.h (created)
[        ]   issue/ios/Runner.xcodeproj/xcshareddata/xcschemes/Runner.xcscheme (created)
[        ]   issue/ios/Runner.xcodeproj/project.pbxproj (created)
[   +6 ms]   issue/ios/.gitignore (created)
[        ]   issue/ios/Flutter/Debug.xcconfig (created)
[        ]   issue/ios/Flutter/Release.xcconfig (created)
[        ]   issue/ios/Flutter/AppFrameworkInfo.plist (created)
[        ]   issue/ios/Runner/Assets.xcassets/AppIcon.appiconset/Contents.json (created)
[        ]   issue/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-1024x1024@1x.png (created)
[        ]   issue/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-20x20@1x.png (created)
[        ]   issue/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-20x20@2x.png (created)
[        ]   issue/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-20x20@3x.png (created)
[        ]   issue/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-29x29@1x.png (created)
[        ]   issue/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-29x29@2x.png (created)
[        ]   issue/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-29x29@3x.png (created)
[        ]   issue/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-40x40@1x.png (created)
[        ]   issue/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-40x40@2x.png (created)
[        ]   issue/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-40x40@3x.png (created)
[        ]   issue/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-60x60@2x.png (created)
[        ]   issue/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-60x60@3x.png (created)
[        ]   issue/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-76x76@1x.png (created)
[        ]   issue/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-76x76@2x.png (created)
[        ]   issue/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-83.5x83.5@2x.png (created)
[        ]   issue/ios/Runner/Assets.xcassets/LaunchImage.imageset/Contents.json (created)
[        ]   issue/ios/Runner/Assets.xcassets/LaunchImage.imageset/LaunchImage.png (created)
[        ]   issue/ios/Runner/Assets.xcassets/LaunchImage.imageset/LaunchImage@2x.png (created)
[        ]   issue/ios/Runner/Assets.xcassets/LaunchImage.imageset/LaunchImage@3x.png (created)
[        ]   issue/ios/Runner/Assets.xcassets/LaunchImage.imageset/README.md (created)
[        ]   issue/ios/Runner/Base.lproj/LaunchScreen.storyboard (created)
[        ]   issue/ios/Runner/Base.lproj/Main.storyboard (created)
[        ]   issue/ios/Runner/Info.plist (created)
[        ]   issue/ios/Runner.xcodeproj/project.xcworkspace/contents.xcworkspacedata (created)
[        ]   issue/ios/Runner.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist (created)
[        ]   issue/ios/Runner.xcodeproj/project.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings (created)
[        ]   issue/ios/Runner.xcodeproj/xcshareddata/xcschemes/Runner.xcscheme (existing - skipped)
[        ]   issue/ios/Runner.xcworkspace/contents.xcworkspacedata (created)
[        ]   issue/ios/Runner.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist (created)
[        ]   issue/ios/Runner.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings (created)
[        ]   issue/lib/main.dart (created)
[   +4 ms]   issue/linux/.gitignore (created)
[        ]   issue/linux/CMakeLists.txt (created)
[   +2 ms]   issue/linux/flutter/CMakeLists.txt (created)
[        ]   issue/linux/my_application.h (created)
[        ]   issue/linux/main.cc (created)
[        ]   issue/linux/my_application.cc (created)
[        ]   issue/issue.iml (created)
[        ]   issue/test/widget_test.dart (created)
[   +1 ms]   issue/web/favicon.png (created)
[        ]   issue/web/icons/Icon-192.png (created)
[        ]   issue/web/icons/Icon-512.png (created)
[        ]   issue/web/manifest.json (created)
[        ]   issue/web/index.html (created)
[   +1 ms]   issue/.gitignore (created)
[        ]   issue/integration_test/app_test.dart (created)
[        ]   issue/integration_test/driver.dart (created)
[        ]   issue/pubspec.yaml (created)
[  +59 ms] Running "flutter pub get" in issue...
[   +5 ms] executing: [/tmp/issue/a/issue/] /opt/flutter/bin/cache/dart-sdk/bin/pub --verbose get --no-precompile
[  +92 ms] MSG : Resolving dependencies...
[  +13 ms] MSG : + _fe_analyzer_shared 12.0.0 (14.0.0 available)
[        ]     | + analyzer 0.40.6 (0.41.1 available)
[        ]     | + archive 2.0.13
[        ]     | + args 1.6.0
[        ]     | + async 2.5.0-nullsafety.3
[        ]     | + boolean_selector 2.1.0-nullsafety.3
[        ]     | + characters 1.1.0-nullsafety.5
[        ]     | + charcode 1.2.0-nullsafety.3
[        ]     | + cli_util 0.2.0
[        ]     | + clock 1.1.0-nullsafety.3
[        ]     | + collection 1.15.0-nullsafety.5
[        ]     | + convert 2.1.1
[        ]     | + coverage 0.14.2
[        ]     | + crypto 2.1.5
[        ]     | + cupertino_icons 1.0.2
[        ]     | + fake_async 1.2.0-nullsafety.3
[        ]     | + file 6.0.0-nullsafety.4
[        ]     | + flutter 0.0.0 from sdk flutter
[        ]     | + flutter_driver 0.0.0 from sdk flutter
[        ]     | + flutter_test 0.0.0 from sdk flutter
[        ]     | + fuchsia_remote_debug_protocol 0.0.0 from sdk flutter
[        ]     | + glob 1.2.0
[        ]     | + integration_test 0.9.2+2 from sdk flutter
[        ]     | + io 0.3.4
[        ]     | + js 0.6.3-nullsafety.3
[        ]     | + json_rpc_2 2.2.2
[        ]     | + logging 0.11.4
[        ]     | + matcher 0.12.10-nullsafety.3
[        ]     | + meta 1.3.0-nullsafety.6
[        ]     | + node_interop 1.2.1
[        ]     | + node_io 1.1.1 (1.2.0 available)
[        ]     | + package_config 1.9.3
[        ]     | + path 1.8.0-nullsafety.3
[        ]     | + pedantic 1.10.0-nullsafety.3
[        ]     | + platform 3.0.0-nullsafety.4
[        ]     | + pool 1.5.0-nullsafety.3
[        ]     | + process 4.0.0-nullsafety.4
[        ]     | + pub_semver 1.4.4
[        ]     | + sky_engine 0.0.99 from sdk flutter
[        ]     | + source_map_stack_trace 2.1.0-nullsafety.4
[        ]     | + source_maps 0.10.10-nullsafety.3
[  +12 ms]     | + source_span 1.8.0-nullsafety.4
[        ]     | + stack_trace 1.10.0-nullsafety.6
[        ]     | + stream_channel 2.1.0-nullsafety.3
[        ]     | + string_scanner 1.1.0-nullsafety.3
[        ]     | + sync_http 0.2.0
[        ]     | + term_glyph 1.2.0-nullsafety.3
[        ]     | + test_api 0.2.19-nullsafety.6
[        ]     | + test_core 0.3.12-nullsafety.9 (0.3.12-nullsafety.12 available)
[        ]     | + typed_data 1.3.0-nullsafety.5
[        ]     | + vector_math 2.1.0-nullsafety.5
[   +6 ms]     | + vm_service 5.5.0
[        ]     | + watcher 0.9.7+15
[        ]     | + web_socket_channel 1.1.0
[        ]     | + webdriver 2.1.2
[        ]     | + yaml 2.2.1
[   +1 ms] MSG : Changed 56 dependencies!
[  +24 ms] Running "flutter pub get" in issue... (completed in 2,929ms)
[ +116 ms] Found plugin integration_test at /opt/flutter/packages/integration_test/
[ +182 ms] Found plugin integration_test at /opt/flutter/packages/integration_test/
[   +6 ms] Generating /tmp/issue/a/issue/android/app/src/main/java/io/flutter/plugins/GeneratedPluginRegistrant.java
[  +10 ms] Wrote 86 files.
[        ]
           All done!
[   +1 ms] In order to run your application, type:

             $ cd issue
             $ flutter run

           Your application code is in issue/lib/main.dart.

[        ] "flutter create" took 3,665ms.
[  +14 ms] ensureAnalyticsSent: 9ms
[   +2 ms] Running shutdown hooks
[        ] Shutdown hooks complete
[        ] exiting with code 0
```

```
[✓] Flutter (Channel dev, 1.26.0-1.0.pre, on Linux, locale en_US.UTF-8)
    • Flutter version 1.26.0-1.0.pre at /opt/flutter
    • Framework revision 63062a6443 (3 weeks ago), 2020-12-13 23:19:13 +0800
    • Engine revision 4797b06652
    • Dart version 2.12.0 (build 2.12.0-141.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /opt/android-sdk
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = /opt/android-sdk
    • Java binary at: /home/adabru/work/armin/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✗] Chrome - develop for the web (Cannot find Chrome executable at google-chrome)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[✓] Linux toolchain - develop for Linux desktop
    • clang version 11.0.0
    • cmake version 3.19.2
    • ninja version 1.10.2
    • pkg-config version 1.7.3

[✓] Android Studio (version 3.6)
    • Android Studio at /home/adabru/work/armin/android-studio
    • Flutter plugin version 49.0.1
    • Dart plugin version 192.8052
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

[✓] Connected device (1 available)
    • Linux (desktop) • linux • linux-x64 • Linux

! Doctor found issues in 1 category.
```

</details>

**Comment by shyndman** [comment]
Hey @adabru,

Great issue! This would be useful, and is sort of the other side of #28737.

I _think_ (without having tried it) that [drag_target.dart#L708](https://github.com/flutter/flutter/blob/1.26.0-1.0.pre/packages/flutter/lib/src/widgets/drag_target.dart#L708) may be responsible for what you're seeing on the drop side. `HitTestBehavior.translucent` indicates that hits should be reported on this `DragTarget` anytime the mouse pointer is inside the widget's bounds.

If it were changed to `HitTestBehavior.deferToChild`, I think it will work for your use case, by delegating the `hitTest` logic to the subtree.

**Comment by shyndman** [comment]
I gave it a shot by modifying my local Flutter's source, and sure enough, changing the MetaData's behavior to `HitTestBehavior.deferToChild` works in the way you want:
https://user-images.githubusercontent.com/42326/103721333-69e3e400-4f9b-11eb-9622-13707fb19253.mov

Not that this helps you much, but maybe the Flutter folks could be convinced to make this behavior configurable. 🤞

**Comment by pedromassangocode** [comment]
I was able to reproduce it.
I would consider this as proposal since this can be fixed by exposing the `hitTestBehavior` of the `DragTarget`.

![gif](https://user-images.githubusercontent.com/67424416/103742658-7d269c00-4ffb-11eb-964c-b0c8a2dc5809.gif)

<details><summary>flutter doctor</summary>

```
[✓] Flutter (Channel master, 1.26.0-2.0.pre.200, on Mac OS X 10.15.7 19H2 darwin-x64, locale en)
    • Flutter version 1.26.0-2.0.pre.200 at /Users/pedromassango/dev/SDKs/flutter_master
    • Framework revision d0e9959a46 (4 hours ago), 2021-01-05 22:14:03 -0500
    • Engine revision e115066dcd
    • Dart version 2.12.0 (build 2.12.0-179.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/pedromassango/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS (Xcode 12.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.1, Build version 12A7403
    ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] IntelliJ IDEA Community Edition (version 2020.3.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 52.1.5
    • Dart plugin version 203.6912

[✓] VS Code (version 1.51.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.16.0

[✓] Connected device (3 available)
    • sdk gphone x86 64 (mobile) • emulator-5554 • android-x64    • Android 11 (API 30) (emulator)
    • macOS (desktop)            • macos         • darwin-x64     • Mac OS X 10.15.7 19H2 darwin-x64
    • Chrome (web)               • chrome        • web-javascript • Google Chrome 87.0.4280.88

! Doctor found issues in 1 category.
Process finished with exit code 0
```
</details>

**Comment by adabru** [comment]
Hi @shyndman thank you very much for your investigation! Yeah I looked into the hitTest function before but the MetaData widget seemed so insignificant to me that I overlooked its `behaviour` property ;-) 

I think your solution does help me. I had a similiar issue with MouseRegion and I just copied the flutter-code for MouseRegion to a local class (~150 lines of code) with the problematic line changed. I intend to do the same thing with the DragTarget code and change the line you mentioned. That's another advantage of the framework being open source.
